runtime/compiler/optimizer/TreeLowering.cpp

```diff
@@ -719,3 +756,719 @@ TR::TreeLowering::lowerArrayStoreCHK(TR::Node *node, TR::TreeTop *tt)
       cfg->addEdge(prevBlock, arrayStoreCheckBlock);
       }
    }
+
+static TR::Node *
+saveGlRegDepsFromBBExit(TR::Node *bbExitNode, TR::Compilation *comp, bool enableTrace, TR_GlobalRegisterNumber registerToSkip = -1)
```

Comment:
The name suggests that this function will save the `GlRegDeps` somewhere.  How about something like `cloneAndTweakGlRegDepsFromBBExit` or something like that?  Something to suggest that the node that's returned will almost be a copy, but not quite.